features/cellular/framework/AT/AT_CellularNetwork.cpp

```diff
@@ -1319,3 +1340,50 @@ nsapi_error_t AT_CellularNetwork::get_operator_names(operator_names_list &op_nam
     _at.resp_stop();
     return _at.unlock_return_error();
 }
+
+nsapi_error_t AT_CellularNetwork::get_active_context(char *apn, nsapi_ip_stack_t stack_type)
+{
+    _at.lock();
+
+    if (apn) {
+        _apn = apn;
+    }
+    if (stack_type != DEFAULT_STACK) {
+        _ip_stack_type_requested = stack_type;
+    }
+
+    int cid_active[PDP_CONTEXT_COUNT];
+    int cid_count = 0;
+
+    // read active contexts
+    _at.cmd_start("AT+CGACT?");
+    _at.cmd_stop();
+    _at.resp_start("+CGACT:");
+    while (_at.info_resp()) {
+        int context_id = _at.read_int();
+        if (_at.read_int() == 1) {
+            cid_active[cid_count] = context_id;
+            cid_count++;
+            if (cid_count >= PDP_CONTEXT_COUNT) {
+                break;
+            }
+        }
+    }
+    _at.resp_stop();
+
+    // try to find a suitable context from active ones
+    int cid = -1;
+    for (int i = 0; i < cid_count; i++) {
+        if (get_context(cid_active[i])) {
+            _is_context_active = true;
+            cid = cid_active[i];
+            tr_debug("Found active context %d", cid);
+            break;
+        }
+    }
+
+     _at.unlock();
+
+    return _is_context_active ? cid : NSAPI_ERROR_NO_CONNECTION;
```

Comment:
Should we return nsapi_error_t e.g. : return _is_context_active ? NSAPI_ERROR_OK : NSAPI_ERROR_NO_CONNECTION;
and give cid as reference?